File: Python/739_Daily_Temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        ans = [0]*len(T)
        stack = []
        
        for i,val in enumerate(T):
            while stack and T[stack[-1]]<val:
                cur = stack.pop()
                ans[cur] = i - cur
            stack.append(i)
        return ans

File: C++/739_Daily_Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        vector<int> res(T.size(),0);
        vector<int> stack;
        for(int i=0;i<T.size();i++)
        {
            while(stack.empty()!=1 && T[stack.back()]<T[i])
            {
                int cur = stack.back();
                res[cur] = i-cur;
                stack.pop_back();
            }
            stack.push_back(i);
        }
        return res;

    }
};
